robot-server/robot_server/modules/module_data_mapper.py

```diff
@@ -54,10 +55,23 @@ def map_data(
         # rely on Pydantic to check/coerce data fields from dicts at run time
         if module_type == ModuleType.MAGNETIC:
             module_cls = MagneticModule
+
+            live_data_height = live_data["data"].get("height")
+            assert isinstance(
+                live_data_height, (int, float)
+            ), f"Expected magnetic module height, got {live_data_height}"
+
+            # Origin of height reported by hardware API is the magnet home
+            # Origin we report to the user should be labware bottom
+            # Also, magnetic module v1 reports height in half millimeters
+            height_from_base = live_data_height - OFFSET_TO_LABWARE_BOTTOM[model]
+            if module_model == ModuleModel.MAGNETIC_MODULE_V1:
+                height_from_base = height_from_base / 2
```

Comment:
I don't have a strong opinion either way, but would this be more readable?

```suggestion
                height_from_base /= 2
```